velox/exec/NestedLoopJoinProbe.cpp

```diff
@@ -401,6 +403,11 @@ RowVectorPtr NestedLoopJoinProbe::doMatch(vector_size_t probeCnt) {
   VELOX_CHECK(!hasProbedAllBuildData());
 
   if (joinCondition_ == nullptr) {
+    // All rows in SelectivityVector for probe and build must be marked valid
+    // for a CrossJoin.
+    probeMatched_.setAll();
```

Comment:
@karteekmurthys : These setAll(...) calls shouldn't be needed if we make the other change to needsProbe/BuildMismatch or add a check in buildMismatchedOutput.